src/script/lua_api/l_object.cpp

```diff
@@ -544,14 +551,122 @@ int ObjectRef::l_get_bone_position(lua_State *L)
 		return 0;
 
 	std::string bone = readParam<std::string>(L, 2, "");
+	const BoneOverride *props = sao->getBoneOverride(bone);
+	push_v3f(L, props->position.vector);
+	v3f euler_rot;
+	props->rotation.next.toEuler(euler_rot);
+	push_v3f(L, euler_rot * core::RADTODEG);
+	return 2;
+}
 
-	v3f position = v3f(0, 0, 0);
-	v3f rotation = v3f(0, 0, 0);
-	sao->getBonePosition(bone, &position, &rotation);
+// set_bone_override(self, bone, override)
+int ObjectRef::l_set_bone_override(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	ObjectRef *ref = checkobject(L, 1);
+	ServerActiveObject *sao = getobject(ref);
+	if (sao == NULL)
+		return 0;
 
-	push_v3f(L, position);
-	push_v3f(L, rotation);
-	return 2;
+	std::string bone;
+	if (!lua_isnil(L, 2))
+		bone = readParam<std::string>(L, 2);
+	BoneOverride *props = new BoneOverride();
```

Comment:
When an error is thrown in one of the checks below (`check_v3f`), this pointer is not freed. Same issue for `l_set_bone_position`

Suggestion: Use `std::unique_ptr` + `.release()`

Reply:
Like this? https://github.com/minetest/minetest/pull/12388/commits/931f1b53b25d15c7a0a0739b0df7ca39816c1630

Reply:
`props.get()->` can be shortened to `props->` because `std::unique_ptr` overloads the indirection operator.

Reply:
> `props.get()->` can be shortened to `props->` because `std::unique_ptr` overloads the indirection operator. EDIT: otherwise fine.

I was suspecting something like that. Thanks.